File: LeetCode/Explore/TopInterview/Amazon/SortingSearching/253_meeting_rooms_II.cpp
class Solution {
public:
    int minMeetingRooms(vector<vector<int>>& intervals) {
      int counter = 0;
      vector<int> ap, bp;
      for (auto &interval : intervals) {
        ap.push_back(interval[0]);
        bp.push_back(interval[1]);
      }
      sort(ap.begin(), ap.end());
      sort(bp.begin(), bp.end());
      for (int i = 0, j = 0; i < intervals.size(); i++) {
        if (ap[i] < bp[j]) {
          counter++;
        }
        else {
          j++;
        }
      }
      return counter;
    }
};

File: LeetCode/Explore/TopInterview/Amazon/SortingSearching/253_meeting_rooms_II.py
class Solution:
    def minMeetingRooms(self, intervals: List[List[int]]) -> int:
        ap, bp = list(), list()
        for interval in intervals:
          ap.append(interval[0])
          bp.append(interval[1])
          
        ap.sort()
        bp.sort()
        j = 0
        counter = 0
        
        for i in range(len(intervals)):
          if ap[i] < bp[j]:
            counter += 1
          else:
            j += 1
            
        return counter
